pkg/apis/security/v1alpha1/types.go

```diff
@@ -116,6 +116,12 @@ type Rule struct {
 	// conjunction with NetworkPolicySpec/ClusterNetworkPolicySpec.AppliedTo.
 	// +optional
 	AppliedTo []NetworkPolicyPeer `json:"appliedTo,omitempty"`
+	// SourceGroups is a list of ClusterGroup names which must be set as the
+	// source for this rule.
+	SourceGroups []string `json:"sourceGroups,omitempty"`
```

Comment:
Not sure if this has been discussed. I wonder why SourceGroups and DestinationGroups are not in From and To NetworkPolicyPeer as NetworkPolicyPeer already describes the source/destination of the traffic.

Reply:
moved to peer.. 

initially i thought adding a new field to Peer would ensure the semantics of an empty peer does not change for older clients. but since all the change is isolated in the controller itself, that is no longer a concern.